teuthology/packaging.py

```diff
@@ -936,7 +938,12 @@ def get_sha1(url):
 
     def assert_result(self):
         if len(self._result.json()) == 0:
-            raise VersionNotFoundError(self._result.url)
+            if not self.try_noarch and self.arch != "noarch":
+                self.try_noarch = True
+                self._result_obj = None
```

Comment:
Nit: it's not immediately clear to me how the query will restart. Will it just automatically retry if `_result_obj` is `None` and it receives a `VersionNotFoundError`?

Reply:
The Line#940 will try to expand and assign the `self._result_obj` based the `self.arch`, which is from the remote OS. For our issue it will be the `x86_64` and the url will be:

`https://shaman.ceph.com/api/search/?status=ready&project=ceph-iscsi&flavor=default&distros=centos%2F8%2Fx86_64&sha1=latest`

But the `self._result.json()` length will be 0, before raising the Error it will try to read the `self._result` again in Line#945 bellow which will try to expand and assign it with new url based the `noarch` again when it see the `self._result_obj` is `None`. 

Only when both the `x86_64` and `noarch` fail will it raise the Error.

Because the `assert_result()` checking is in everywhere, so I put the above logic in it. Maybe we should move it to:

```
 861     @property
 862     def _result(self):
 863         if getattr(self, '_result_obj', None) is None:                                                                               
 864             self._result_obj = self._search()   
 865         return self._result_obj
```

???

Reply:
@kshtsk @dillaman 
BTW, will it be better to add an opetion likes `try_noarch = True` to enable trying the alternate noarch uri, if the uri based something like the `x86_64` does not exist ? And as default the `try_noarch = False` or absent.

Reply:
Personally, I think it would be cleaner if we just added a `force_noarch: True` or just `noarch: True` to the config where needed instead of this hidden logic to always implicitly attempt to fallback to `noarch`.

Reply:
Okay, sounds reasonable. Will add it. Thanks.

Reply:
Updated it. Thanks.

Reply:
Sorry for being late. But actually I do not understand why do we need an option for this, what is the bad to always check both repos?

Reply:
It wasn't checking both -- it was checking for x86_64 and failing that, then it checked for noarch. If it adds both repos if both are valid, I'm fine with that -- but this is also a use-case where we know specifically that we want to pull from noarch so why not just offer the override?

Reply:
@dillaman thanks for the explanation, so if we want packaging module should override arch, then we need to provide this withing packaging namespace.
I would like to see also the docs updated to explain how this is supposed to be used from the user stand point since it is kinda new feature.

Reply:
@kshtsk BTW, which doc file should we document it ? I didn't find an appropriate one.

Reply:
I would say `docs/detailed_test_config.rst`.

Reply:
Maybe add somewhere else afterwards.

Reply:
Okay, will document it. Thanks.